### DISCUSSION TITLE
Shelly Plus 2PM template results in negative power meter

### DISCUSSION THREAD

**Discussion by miguelpruivo** [RavenSystem/esp-homekit-devices#2243] [General]
Just probably a quick fix, but the template in the database for the Shelly Plus 2 PM as a double way switch with power meter results in a negative power meter for amps and kWh.

```
{"c":{"io":[[[12,13,0],2],[[4],6,1],[[5,18],6,0,1],[[35],10,0,3],[[27]]],"l":0,"b":[[4,5]],"ic":[[25,26,100]]},"a":[{"0":{"r":[[13]]},"1":{"r":[[13,1]]},"b":[[5],[5,0]],"es":[{"0":{"r":[[12]]},"1":{"r":[[12,1]]},"b":[[18],[18,0]]},{"t":75,"cd":1.5,"n":2,"dt":[0,56],"vf":0.0000382602,"vo":-0.068,"cf":0.00000949523,"co":-0.017,"pf":0.006097560976},{"t":75,"n":3,"dt":[0,56],"vf":0.0000382602,"vo":-0.068,"cf":0.00000949523,"co":-0.017,"pf":0.006097560976},{"t":22,"h":2,"n":5,"g":35,"j":10,"y0":[{"v":75,"r":1,"0":{"m":[[1],[2]]}}]}]}]}
```

**Comment by tonymld** [comment]
Hello! I need a simple template for Shelly 2PM just to use two buttons to turn on/off two lights. No power metering, double or long press needed. just simple on/off for HomeKit.

**Comment by miguelpruivo** [comment]
> Hello! I need a simple template for Shelly 2PM just to use two buttons to turn on/off two lights. No power metering, double or long press needed. just simple on/off for HomeKit.

I'd like that as well but keeping the turn-off when heating safety feature — I'm not sure how impactful the PM can have in the performance for HK reliability, based on the HomeBench it looks like the PM service has worse performance than the switch service.

Maybe @RavenSystem can help us out.

**Comment by RavenSystem** [comment]
If you flashed it with OTA, you must do a complete power cycle un-powering device completely to reboot all internal hardware.
